From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Spottedleaf <spottedleaf@spottedleaf.dev>
Date: Tue, 8 Sep 2020 01:30:49 -0700
Subject: [PATCH] Fix missing block property portal crash

Looks like it appears when the portal isn't created (cancelled
event). So, just use the original direction and call it a day.

diff --git a/src/main/java/net/minecraft/server/BlockPortalShape.java b/src/main/java/net/minecraft/server/BlockPortalShape.java
index 6ef81aeb4c63bc6c23163796dbd977602ca2f540..a848447260f77b85e5f1bfaedeb16d7d068e00d8 100644
--- a/src/main/java/net/minecraft/server/BlockPortalShape.java
+++ b/src/main/java/net/minecraft/server/BlockPortalShape.java
@@ -227,7 +227,7 @@ public class BlockPortalShape {
     public static ShapeDetectorShape a(WorldServer worldserver, BlockUtil.Rectangle blockutil_rectangle, EnumDirection.EnumAxis enumdirection_enumaxis, Vec3D vec3d, EntitySize entitysize, Vec3D vec3d1, float f, float f1, CraftPortalEvent portalEventInfo) { // CraftBukkit // PAIL rename toDetectorShape
         BlockPosition blockposition = blockutil_rectangle.origin;
         IBlockData iblockdata = worldserver.getType(blockposition);
-        EnumDirection.EnumAxis enumdirection_enumaxis1 = (EnumDirection.EnumAxis) iblockdata.get(BlockProperties.E);
+        EnumDirection.EnumAxis enumdirection_enumaxis1 = !iblockdata.contains(BlockProperties.E) ? enumdirection_enumaxis : (EnumDirection.EnumAxis) iblockdata.get(BlockProperties.E); // Tuinity - use provided direction if the block doesn't have one
         double d0 = (double) blockutil_rectangle.side1;
         double d1 = (double) blockutil_rectangle.side2;
         int i = enumdirection_enumaxis == enumdirection_enumaxis1 ? 0 : 90;
